apps/antalmanac/src/components/Map/Map.tsx

```diff
@@ -167,6 +167,22 @@ export default function CourseMap() {
         };
     }, []);
 
+    useEffect(() => {
+        const updateCustomEventMarkers = () => {
+            setCustomEventMarkers(getCustomEventPerBuilding());
+        };
+
+        AppStore.on('customEventsChange', updateCustomEventMarkers);
+        AppStore.on('currentScheduleIndexChange', updateCustomEventMarkers);
+        AppStore.on('colorChange', updateCustomEventMarkers);
+
+        return () => {
+            AppStore.removeListener('customEventsChange', updateCustomEventMarkers);
+            AppStore.removeListener('currentScheduleIndexChange', updateCustomEventMarkers);
+            AppStore.removeListener('colorChange', updateCustomEventMarkers);
+        };
+    }, []);
+
```

Comment:
There's a good amount of code duplication between this `useEffect()` and the one above for regular markers. Instead of adding three new listeners, you can change the original `updateMarkers()` function to _also_ set custom event markers.

Whether you refactor the name of `updateMarkers()` is up to you